File: server/game/character.go
package game

type Position = [3]float64

type Character struct {
	Id          string   `json:"id"`
	HairColor   string   `json:"hairColor"`
	TopColor    string   `json:"topColor"`
	BottomColor string   `json:"bottomColor"`
	Position    Position `json:"position"`
	Rotate      float64  `json:"rotate"`
	Anim        string   `json:"anim"`
}

File: client/src/types/character.ts
import { Position } from './physics';

export type Character = {
  uuid: string;
  hair_color: string;
  top_color: string;
  bottom_color: string;
  position: Position;
  rotate: number;
  anim: CharacterCommonAnimationNames;
};

export type CharacterCommonAnimationNames =
  | 'run'
  | 'run_back'
  | 'run_right'
  | 'run_left'
  | 'idle'
  | 'walk'
  | 'jump'
  | 'wave';

export type ServerCharactersMap = {
  [key: string]: Character;
};

export type MoveCharacterBroadcast = {
  uuid: string;
  position: Position;
  rotate: number;
  anim: CharacterCommonAnimationNames;
};
